lib/controllers/root-controller.js

```diff
@@ -574,6 +587,44 @@ export default class RootController extends React.Component {
     viewElement.focus();
   }
 
+  viewChangesForCurrentFile(stagingStatus) {
+    const editor = this.props.workspace.getActiveTextEditor();
+    const absFilePath = editor.getPath();
+    const repoPath = this.props.repository.getWorkingDirectoryPath();
+    if (absFilePath.startsWith(repoPath)) {
+      const filePath = absFilePath.slice(repoPath.length + 1);
+      this.quietlySelectItem(filePath, stagingStatus);
+      const splitDirection = this.props.config.get('github.viewChangesForCurrentFileDiffPaneSplitDirection');
+      const pane = atom.workspace.getActivePane();
+      if (splitDirection === 'right') {
+        pane.splitRight();
+      } else if (splitDirection === 'down') {
+        pane.splitDown();
+      }
+      return this.diveIntoFilePatchForPath(filePath, stagingStatus, {
+        lineNumber: editor.getCursorBufferPosition().row + 1,
+        amending: stagingStatus === 'staged' && this.state.isAmending,
+      });
+    } else {
+      throw new Error(`${absFilePath} does not belong to repo ${repoPath}`);
```

Comment:
Thinking this through: because the active text editor sets the active repository context, the only time that this could happen is if it's triggered in the window between opening an editor on a file not in the active repository and the render context change completing. Performance captures indicate that window is only in the tens of milliseconds range, so it's unlikely that anyone could hit this manually.

_Eventually_ we might want to add logic to the effect of "wait on a Promise that resolves when the active context is what we expect it to be, then trigger `diveIntoFilePatchForPath`." Not worth holding up an initial merge for though I think.